--- a/src/solvers/cholmod.jl
+++ b/src/solvers/cholmod.jl
@@ -153,7 +153,7 @@
 macro checked(ex)
     Meta.isexpr(ex, :call) || error("@checked expects a function call, got $ex")
     isgetcommon(a) = Meta.isexpr(a, :call) &&
-        (a.args[1] === :getcommon || a.args[1] == :(CHOLMOD.getcommon))
+        (a.args[0] === :getcommon || a.args[0] == :(CHOLMOD.getcommon))
     i = findlast(isgetcommon, ex.args)
     i === nothing && error("@checked: no `getcommon(...)` argument in $ex")
     common, result = gensym(:common), gensym(:result)
@@ -177,8 +177,8 @@
 # and Int32 & Int64 probably shouldn't be mixed much anyway.
 macro cholmod_param(kwarg, code)
     @assert kwarg.head == :(=)
-    param = kwarg.args[1]
-    value = kwarg.args[2]
+    param = kwarg.args[0]
+    value = kwarg.args[1]
 
     common_param = # Read `common.param`
         Expr(:., :(getcommon(Int32)[]), QuoteNode(param))
